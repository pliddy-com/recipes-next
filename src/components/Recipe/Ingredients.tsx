import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';

import RestaurantMenuIcon from '@mui/icons-material/RestaurantMenu';

import { IngredientsDefaultFragment } from 'types/generated/graphql';

const styles = {
  bullets: {
    height: '32px',
  },
};

interface IngredientsProps {
  sections?: (IngredientsDefaultFragment | null)[];
}

const Ingredients = ({ sections }: IngredientsProps) => {
  if (!sections) return null;

  return (
    <Stack direction="column" spacing={1}>
      <Typography variant="h2" gutterBottom>
        Ingredients
      </Typography>
      {sections &&
        sections.map((section) => {
          const { label, ingredientList } = section ?? {};
          const ingredients = ingredientList;

          return (
            <Stack key={label} direction="column" spacing={0}>
              {label !== 'Ingredients' && (
                <Typography variant="h3">{label}</Typography>
              )}
              <List className="recipeList">
                {ingredients &&
                  ingredients.map((ingredient) => (
                    <ListItem key={ingredient} disableGutters>
                      <ListItemIcon>
                        <RestaurantMenuIcon color="secondary" />
                      </ListItemIcon>
                      <ListItemText primary={ingredient} sx={styles.bullets} />
                    </ListItem>
                  ))}
              </List>
            </Stack>
          );
        })}
    </Stack>
  );
};

export default Ingredients;
