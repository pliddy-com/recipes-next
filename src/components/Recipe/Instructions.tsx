import Avatar from '@mui/material/Avatar';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemAvatar from '@mui/material/ListItemAvatar';
import ListItemText from '@mui/material/ListItemText';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';

import { InstructionsDefaultFragment } from 'types/generated/graphql';

interface InstructionsProps {
  sections?: InstructionsDefaultFragment[];
}

const Instructions = ({ sections }: InstructionsProps) => {
  if (!sections) return null;

  // use num to set starting value for each instructionList section
  // so numbering is continuous across list sections
  let num = 1;

  return (
    <Stack direction="column" spacing={1}>
      <Typography variant="h2" gutterBottom>
        Instructions
      </Typography>
      {sections &&
        sections.map((section) => {
          const { label, instructionList } = section ?? {};
          const instructions = instructionList;
          return (
            <Stack key={label} direction="column" spacing={0}>
              <Typography variant="h3">{label}</Typography>

              <List className="recipeList">
                {instructions &&
                  instructions.map((instruction) => (
                    <ListItem key={`instruction-${num}`} disableGutters>
                      <ListItemAvatar>
                        <Avatar color="secondary">{num++}</Avatar>
                      </ListItemAvatar>
                      <ListItemText primary={instruction} />
                    </ListItem>
                  ))}
              </List>
            </Stack>
          );
        })}
    </Stack>
  );
};

export default Instructions;
