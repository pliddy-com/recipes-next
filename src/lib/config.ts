interface PageProps {
  defaultDescription?: string;
  defaultTitle?: string;
  description?: string;
  domain?: string;
  title?: string;
}

interface MicrocopyProps {
  category?: PageProps;
  index?: PageProps;
  recipe?: PageProps;
  site?: PageProps;
  tag?: PageProps;
}

export interface ConfigProps {
  microcopy?: MicrocopyProps;
}

const microcopy: MicrocopyProps = {
  category: {
    defaultDescription: 'A collection of recipies categorized as',
    defaultTitle: 'Category',
  },
  index: {
    defaultTitle: 'All Recipes',
    description: "Patrick's recipe collection",
  },
  recipe: {
    defaultDescription: 'A recipe for',
    defaultTitle: 'Recipe',
  },
  site: {
    title: "Patrick's Recipes",
    domain: 'https://recipes.pliddy.com',
  },
  tag: {
    defaultDescription: 'A collection of recipies tagged as',
    defaultTitle: 'Tag',
  },
};

const config: ConfigProps = {
  microcopy,
};

export default config;
