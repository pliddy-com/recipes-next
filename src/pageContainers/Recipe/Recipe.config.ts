interface Breakpoints {
  viewMin?: number;
  imgWidth: number;
}

interface Config {
  breakpoints?: Breakpoints[];
}

const config: Config = {
  breakpoints: [
    {
      viewMin: 704,
      imgWidth: 832,
    },
    {
      viewMin: 448,
      imgWidth: 624,
    },
    {
      imgWidth: 416,
    },
  ],
};

export default config;
