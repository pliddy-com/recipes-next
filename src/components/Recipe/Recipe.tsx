import Image from 'next/image';

import useMediaQuery from '@mui/material/useMediaQuery';
import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
import Grid from '@mui/material/Grid';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';

import Equipment from 'components/Recipe/Equipment';
import Ingredients from 'components/Recipe/Ingredients';
import Instructions from 'components/Recipe/Instructions';
// import Loading from 'components/Loading';
import Notes from 'components/Recipe/Notes';
import Tags from 'components/Recipe/Tags';

// TODO: componentize
import { documentToReactComponents } from '@contentful/rich-text-react-renderer';

import { RecipeDefaultFragment } from 'types/generated/graphql';

import theme from 'theme';

const imgSizes = {
  height: {
    xs: 426,
    sm: 636,
    md: 426,
    lg: 426,
  },
  width: {
    xs: 568,
    sm: 848,
    md: 568,
    lg: 568,
  },
};

interface RecipeProps {
  recipe?: RecipeDefaultFragment;
}

const Recipe = ({ recipe }: RecipeProps) => {
  //   const { slug } = useParams() ?? {};

  const isMd = useMediaQuery(theme.breakpoints.up('md'));
  const isSm = useMediaQuery(theme.breakpoints.up('sm'));
  const size = isMd ? 'md' : isSm ? 'sm' : 'xs';

  if (!recipe) return null;

  const {
    title,
    description,
    equipment,
    image,
    ingredientsCollection,
    instructionsCollection,
    notes,
    tagsCollection,
  } = recipe ?? {};

  const { json: descriptionDoc } = description ?? {};
  const { url, description: imgAlt } = image ?? {};
  const src = url || '';
  const alt = imgAlt || '';

  const { items: ingredientsSections } = ingredientsCollection ?? {};
  const { items: instructionsSections } = instructionsCollection ?? {};
  const { items: tags } = tagsCollection ?? {};

  /*
    contentType: "image/jpeg"
    description: "Sole Meunière on a bed of chopped green lettuce and roma tomatoes with a white balsamic-lemon vinaigrette."
    fileName: "sole-meuniere.jpg"
    height: 3024
    size: 2758954
    title: "Sole Meunière"
    url: "https://images.ctfassets.net/fo9qwg6zarbt/7JzbZn07fY62akLYWhNqDg/df3f00e87d2ebab4f807caac8e2befcb/sole-meuniere.jpg"
    width: 4032
  */

  console.log(image);

  return (
    <Container className="main" component="main">
      <Grid container direction="row" spacing={2}>
        <Grid item md>
          <Typography variant="h1" gutterBottom>
            {title}
          </Typography>
          {descriptionDoc && (
            <Box>{documentToReactComponents(descriptionDoc)}</Box>
          )}
        </Grid>
        {image && (
          <Grid item md>
            <Image
              alt={alt}
              height={imgSizes.height[size]}
              priority={true}
              src={`${src}?h=${imgSizes.height[size]}&fm=webp`}
              style={{ maxWidth: '100%', height: 'auto' }}
              width={imgSizes.width[size]}
            />
          </Grid>
        )}
      </Grid>
      <Stack direction="column" spacing={3}>
        {ingredientsCollection &&
          ingredientsCollection.items &&
          ingredientsSections && <Ingredients sections={ingredientsSections} />}

        {equipment && <Equipment equipment={equipment} />}

        {instructionsSections && (
          <Instructions sections={instructionsSections} />
        )}

        {notes && <Notes notes={notes} />}

        {tags && <Tags tags={tags} />}

        {/* <pre>{JSON.stringify(tags, null, 2)}</pre> */}
      </Stack>
    </Container>
  );
};

export default Recipe;
