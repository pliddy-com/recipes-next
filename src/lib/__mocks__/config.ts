import { ConfigProps } from 'lib/config';

const microcopy = {
  category: {
    defaultDescription: 'Category default description',
    defaultTitle: 'Category Default Title',
  },
  index: {
    defaultTitle: 'Index Default Title',
    description: 'Index default description.',
  },
  recipe: {
    defaultDescription: 'Recipe default description',
    defaultTitle: 'Recipe Default Title',
  },
  site: {
    title: 'Site Title',
  },
  tag: {
    defaultDescription: 'Tag default description',
    defaultTitle: 'Tag Default Title',
  },
};

const config: ConfigProps = {
  microcopy,
};

module.exports = config;
