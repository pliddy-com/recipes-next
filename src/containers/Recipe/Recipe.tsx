import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
// import Grid from '@mui/material/Grid';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';

import DynamicImage from 'components/DynamicImage/DynamicImage';
import Equipment from 'containers/Recipe/Equipment/Equipment';
import Ingredients from 'containers/Recipe/Ingredients/Ingredients';
import Instructions from 'containers/Recipe/Instructions/Instructions';
import Notes from 'containers/Recipe/Notes/Notes';
import RichText from 'components/RichText/RichText';
import Tags from 'containers/Recipe/Tags/Tags';

import {
  RecipeDefaultFragment,
  RecipeDescription,
} from 'types/generated/graphql';

import config from 'lib/config';

interface RecipeProps {
  content?: RecipeDefaultFragment;
}

const Recipe = ({ content }: RecipeProps) => {
  if (!content) return null;

  const {
    title,
    description,
    equipment,
    image,
    ingredientsCollection,
    instructionsCollection,
    notes,
    tagsCollection,
  } = content ?? {};

  const {
    images: {
      props: { recipe: imageProps },
    },
  } = config ?? {};

  // TODO: pass full description text into rich text component
  //       & destructure json & links inside component (needs types for json & links)
  //       Need to remove typecasting of RecipeDescription

  // const { json, links } = description ?? {};

  const { items: ingredientsSections } = ingredientsCollection ?? {};
  const { items: instructionsSections } = instructionsCollection ?? {};
  const { items: tags } = tagsCollection ?? {};

  return (
    <Container className="page">
      <Box>
        <Typography variant="h1" gutterBottom>
          {title}
        </Typography>

        {description && (
          <RichText content={description as RecipeDescription} />
          // <RichText content={description} />
        )}
      </Box>

      {image && (
        <Box mb={4} mt={4} ml={2} mr={2}>
          <DynamicImage image={image} props={imageProps} preload={true} />
        </Box>
      )}

      <Stack direction="column" spacing={3}>
        {ingredientsCollection &&
          ingredientsCollection.items &&
          ingredientsSections && <Ingredients sections={ingredientsSections} />}

        {equipment && <Equipment equipment={equipment} />}

        {instructionsSections && (
          <Instructions sections={instructionsSections} />
        )}

        {notes && <Notes notes={notes} />}

        {tags && <Tags tags={tags} />}

        {/* <pre>{JSON.stringify(tags, null, 2)}</pre> */}
      </Stack>
    </Container>
  );
};

export default Recipe;
