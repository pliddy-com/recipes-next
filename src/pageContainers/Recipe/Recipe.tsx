import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';

import DynamicImage from 'components/DynamicImage/DynamicImage';
import EquipmentSection from 'components/EquipmentSection/EquipmentSection';
import IngredientsSection from 'components/IngredientsSection/IngredientsSection';
import InstructionsSection from 'components/InstructionsSection/InstructionsSection';
import NotesSection from 'components/NotesSection/NotesSection';
import RichText from 'components/RichText/RichText';
import TagsSection from 'components/TagsSection/TagsSection';

import {
  RecipeDefaultFragment,
  RecipeDescription,
} from 'types/generated/graphql';

import config from './Recipe.config';

interface RecipeProps {
  content?: RecipeDefaultFragment;
}

const Recipe = ({ content }: RecipeProps) => {
  const {
    title,
    description,
    equipment,
    image,
    ingredientsCollection,
    instructionsCollection,
    notes,
    tagsCollection,
  } = content ?? {};

  const { breakpoints } = config;

  // TODO: pass full description text into rich text component
  //       & destructure json & links inside component (needs types for json & links)
  //       Need to remove typecasting of RecipeDescription

  // const { json, links } = description ?? {};

  const { items: ingredientsSections } = ingredientsCollection ?? {};
  const { items: instructionsSections } = instructionsCollection ?? {};
  const { items: tags } = tagsCollection ?? {};

  return content ? (
    <Container className="page">
      <Box>
        <Typography variant="h1" gutterBottom>
          {title}
        </Typography>

        {description && (
          <RichText content={description as RecipeDescription} />
          // <RichText content={description} />
        )}
      </Box>
      {image && breakpoints && (
        <Box
          mx={{
            xs: 0,
            sm: 1,
            md: 2,
          }}
          my={4}
        >
          <DynamicImage
            image={image}
            breakpoints={breakpoints}
            preload={true}
          />
        </Box>
      )}
      <Stack direction="column" spacing={3}>
        {ingredientsCollection &&
          ingredientsCollection.items &&
          ingredientsSections && (
            <IngredientsSection sections={ingredientsSections} />
          )}

        {equipment && <EquipmentSection equipment={equipment} />}

        {instructionsSections && (
          <InstructionsSection sections={instructionsSections} />
        )}

        {notes && <NotesSection notes={notes} />}

        {tags && <TagsSection tags={tags} />}

        {/* <pre>{JSON.stringify(tags, null, 2)}</pre> */}
      </Stack>
    </Container>
  ) : null;
};

export default Recipe;
