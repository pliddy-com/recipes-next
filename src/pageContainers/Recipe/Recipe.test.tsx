// import testing-library methods
import { render } from '@testing-library/react';

// add custom jest matchers from jest-dom
import '@testing-library/jest-dom';

// import the component to test
import Recipe from './Recipe';

import { RecipeDefaultFragment } from 'types/generated/graphql';

import config from './Recipe.config';

jest.createMockFromModule('./Recipe.config');

describe('Recipe', () => {
  afterEach(() => {
    jest.resetModules();
  });

  describe('when there is page content', () => {
    it('it renders the RecipeGrid', () => {
      const content = {
        sys: {
          id: 'sysid-0',
          __typename: 'Sys',
        },
        __typename: 'Recipe',
        title: 'Recipe Title',
        slug: 'recipe-title',
        description: {
          json: {
            data: {},
            content: [
              {
                data: {},
                content: [
                  {
                    data: {},
                    marks: [],
                    value: 'Description rich text',
                    nodeType: 'text',
                  },
                ],
                nodeType: 'paragraph',
              },
            ],
            nodeType: 'document',
          },
          links: {
            assets: {
              block: [],
              hyperlink: [],
              __typename: 'RecipeDescriptionAssets',
            },
            entries: {
              inline: [],
              hyperlink: [],
              __typename: 'RecipeDescriptionEntries',
            },
            __typename: 'RecipeDescriptionLinks',
          },
          __typename: 'RecipeDescription',
        },
        abstract: 'Recipe abstract',
        image: {
          sys: {
            id: 'sysid-1',
            __typename: 'Sys',
          },
          __typename: 'Asset',
          title: 'Image Title',
          description: 'Image description.',
          contentType: 'image/jpeg',
          fileName: 'filename.jpg',
          size: 99999,
          url: 'https://recipes.pliddy.com/url',
          height: 300,
          width: 400,
        },
        ingredientsCollection: {
          items: [
            {
              sys: {
                id: 'sysid-2',
                __typename: 'Sys',
              },
              title: 'Ingredients Title',
              slug: 'ingredients-title',
              label: 'Ingredients',
              ingredientList: ['ingredient 1', 'ingredient 2'],
              __typename: 'IngredientSection',
            },
          ],
          __typename: 'RecipeIngredientsCollection',
        },
        equipment: ['equipment 1', 'equipment 2'],
        instructionsCollection: {
          items: [
            {
              sys: {
                id: 'sysid-3',
                __typename: 'Sys',
              },
              title: 'Instructions Title',
              slug: 'instructions-title',
              label: 'Label',
              instructionList: ['Instructions item 1'],
              __typename: 'InstructionSection',
            },
          ],
          __typename: 'RecipeInstructionsCollection',
        },
        notes: ['note 1'],
        tagsCollection: {
          items: [
            {
              sys: {
                id: 'sysid-4',
                __typename: 'Sys',
              },
              __typename: 'Tag',
              title: 'Tag 1',
              slug: 'tag-1',
            },
          ],
          __typename: 'RecipeTagsCollection',
        },
      };

      const { container } = render(
        <Recipe content={content as RecipeDefaultFragment} />
      );

      // test if content is rendered
      const recipe = document.querySelector('.page');
      expect(recipe).toBeInTheDocument();

      // assert that the component matches the existing snapshot
      expect(container).toMatchSnapshot();
    });
  });

  describe('when there is no page content', () => {
    // before each test, delete images node from config
    beforeEach(() => {
      delete config.breakpoints;
    });

    it('it does not render the Recipe', () => {
      const content = undefined;

      render(<Recipe content={content} />);

      // test if content is not rendered
      render(<Recipe content={content} />);

      const recipe = document.querySelector('.page');
      expect(recipe).toBeNull();
    });
  });
});
