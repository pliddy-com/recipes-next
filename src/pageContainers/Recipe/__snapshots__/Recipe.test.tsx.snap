// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe when there is page content it renders the RecipeGrid 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg page css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom css-a4d0r7-MuiTypography-root"
      >
        Recipe Title
      </h1>
      <div
        class="MuiBox-root css-0"
        data-testid="richtext"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-z2eky3-MuiTypography-root"
        >
          Description rich text
        </p>
      </div>
    </div>
    <div
      class="MuiBox-root css-mc0pi9"
    >
      <link
        as="image"
        href="https://recipes.pliddy.com/url"
        imagesizes="900px"
        imagesrcset="https://recipes.pliddy.com/url?w=900&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1800&fm=webp&q=75 2x"
        media="(min-width: 831px)"
        rel="preload"
      />
      <link
        as="image"
        href="https://recipes.pliddy.com/url"
        imagesizes="750px"
        imagesrcset="https://recipes.pliddy.com/url?w=750&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1500&fm=webp&q=75 2x"
        media="(min-width: 665px) and (max-width: 830px)"
        rel="preload"
      />
      <link
        as="image"
        href="https://recipes.pliddy.com/url"
        imagesizes="600px"
        imagesrcset="https://recipes.pliddy.com/url?w=600&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1200&fm=webp&q=75 2x"
        media="(min-width: 483px) and (max-width: 664px)"
        rel="preload"
      />
      <link
        as="image"
        href="https://recipes.pliddy.com/url"
        imagesizes="450px"
        imagesrcset="https://recipes.pliddy.com/url?w=450&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=900&fm=webp&q=75 2x"
        media="(min-width: 333px) and (max-width: 482px)"
        rel="preload"
      />
      <link
        as="image"
        href="https://recipes.pliddy.com/url"
        imagesizes="300px"
        imagesrcset="https://recipes.pliddy.com/url?w=300&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=600&fm=webp&q=75 2x"
        media="(max-width: 332px)"
        rel="preload"
      />
      <picture>
        <source
          media="(min-width: 831px)"
          sizes="900px"
          srcset="https://recipes.pliddy.com/url?w=900&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1800&fm=webp&q=75 2x"
        />
        <source
          media="(min-width: 665px) and (max-width: 830px)"
          sizes="750px"
          srcset="https://recipes.pliddy.com/url?w=750&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1500&fm=webp&q=75 2x"
        />
        <source
          media="(min-width: 483px) and (max-width: 664px)"
          sizes="600px"
          srcset="https://recipes.pliddy.com/url?w=600&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1200&fm=webp&q=75 2x"
        />
        <source
          media="(min-width: 333px) and (max-width: 482px)"
          sizes="450px"
          srcset="https://recipes.pliddy.com/url?w=450&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=900&fm=webp&q=75 2x"
        />
        <source
          media="(max-width: 332px)"
          sizes="300px"
          srcset="https://recipes.pliddy.com/url?w=300&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=600&fm=webp&q=75 2x"
        />
        <img
          alt="Image description."
          height="300"
          loading="eager"
          src="https://recipes.pliddy.com/url"
          style="max-width: 100%; height: auto;"
          width="400"
        />
      </picture>
    </div>
    <div
      class="css-1lwbda4-MuiStack-root"
    >
      <div
        class="css-1okj3ks-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-i1hrn3-MuiTypography-root"
        >
          Ingredients
        </h2>
        <div
          class="css-nen11g-MuiStack-root"
        >
          <ul
            class="MuiList-root MuiList-padding recipeList css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-kw0y8f-MuiSvgIcon-root"
                  data-testid="RestaurantMenuIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m8.1 13.34 2.83-2.83L3.91 3.5c-1.56 1.56-1.56 4.09 0 5.66l4.19 4.18zm6.78-1.81c1.53.71 3.68.21 5.27-1.38 1.91-1.91 2.28-4.65.81-6.12-1.46-1.46-4.2-1.1-6.12.81-1.59 1.59-2.09 3.74-1.38 5.27L3.7 19.87l1.41 1.41L12 14.41l6.88 6.88 1.41-1.41L13.41 13l1.47-1.47z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  ingredient 1
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-kw0y8f-MuiSvgIcon-root"
                  data-testid="RestaurantMenuIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m8.1 13.34 2.83-2.83L3.91 3.5c-1.56 1.56-1.56 4.09 0 5.66l4.19 4.18zm6.78-1.81c1.53.71 3.68.21 5.27-1.38 1.91-1.91 2.28-4.65.81-6.12-1.46-1.46-4.2-1.1-6.12.81-1.59 1.59-2.09 3.74-1.38 5.27L3.7 19.87l1.41 1.41L12 14.41l6.88 6.88 1.41-1.41L13.41 13l1.47-1.47z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  ingredient 2
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="css-nen11g-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
        >
          Equipment
        </h2>
        <ul
          class="MuiList-root MuiList-padding recipeList css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                data-testid="KitchenTwoToneIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5h2v3H8zm0 7h2v5H8zm-2 8h12v-9.02H6V20zm2-8h2v5H8v-5zM6 9h12V4H6v5zm2-4h2v3H8V5z"
                  opacity=".3"
                />
                <path
                  d="M18 2.01 6 2a2 2 0 0 0-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.11-.9-1.99-2-1.99zM18 20H6v-9.02h12V20zm0-11H6V4h12v5zM8 5h2v3H8zm0 7h2v5H8z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                equipment 1
              </span>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                data-testid="KitchenTwoToneIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5h2v3H8zm0 7h2v5H8zm-2 8h12v-9.02H6V20zm2-8h2v5H8v-5zM6 9h12V4H6v5zm2-4h2v3H8V5z"
                  opacity=".3"
                />
                <path
                  d="M18 2.01 6 2a2 2 0 0 0-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.11-.9-1.99-2-1.99zM18 20H6v-9.02h12V20zm0-11H6V4h12v5zM8 5h2v3H8zm0 7h2v5H8z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                equipment 2
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="css-1okj3ks-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-i1hrn3-MuiTypography-root"
        >
          Instructions
        </h2>
        <div
          class="css-nen11g-MuiStack-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
          >
            Label
          </h3>
          <ul
            class="MuiList-root MuiList-padding recipeList css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
            >
              <div
                class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
                  color="secondary"
                >
                  1
                </div>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Instructions item 1
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
        >
          Notes
        </h2>
        <ul>
          <li>
            note 1
          </li>
        </ul>
      </div>
      <div
        class="MuiBox-root css-0"
        data-testid="tags-section"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-i1hrn3-MuiTypography-root"
        >
          Tags
        </h2>
        <div
          class="css-180l46g-MuiStack-root"
          data-testid="tag-buttons"
        >
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-wmndcn-MuiButtonBase-root-MuiButton-root"
            href="/tag/tag-1"
            role="button"
            tabindex="0"
          >
             Tag 1 
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
